src/lsm/lsm_tree.c

```diff
@@ -15,15 +15,49 @@ static int __lsm_tree_open(
 static int __lsm_tree_set_name(WT_SESSION_IMPL *, WT_LSM_TREE *, const char *);
 
 /*
+ * __lsm_tree_meta_discard --
+ *	Free the metadata related LSM tree pointers.
```

Comment:
I'm too familiar with thinking about metadata as the content of the metadata table - could you call this function something else? Maybe `__lsm_tree_discard_state`?